### DISCUSSION TITLE
is it possible to ignore columns when insert

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [diesel-rs/diesel#3007] [Q&A]
I am using diesel to do an insert, this is my code:

```
pub fn add_domain(request: &Json<AddDomainRequest>, login_user_info: LoginUserInfo) {
    let connection = config::establish_connection();
    let timestamp: i64 = get_current_millisecond();
    let new_domain = Domain {
        id: 0,
        domain_name: request.domainName.to_string(),
        domain_url: request.domainUrl.to_string(),
        created_time: timestamp,
        updated_time: timestamp,
        cron: Some("* */1 * * * *".parse().unwrap()),
        next_trigger_time: None,
        monitor_status: None,
        user_id: Option::from(login_user_info.userId),
        expire_date: None,
        days_before_trigger: 7,
        notify_trigger_date: None,
        expire_date_ms: None,
    };
    diesel::insert_into(crate::model::diesel::dolphin::dolphin_schema::domain::table)
        .values(&new_domain)
        .on_conflict_do_nothing()
        .execute(&connection)
        .unwrap();
}

```

but it shows error:

`thread 'rocket-worker-thread' panicked at 'called `Result::unwrap()` on an `Err` value: DatabaseError(__Unknown, "cannot insert into column \"id\"")', src/service/app/cernitor/domain/domain_service.rs:46:10
`

what should I do to ignore the id column and make database auto generate?

**Comment by weiznich** [comment]
Please provide at least the following information:

* The corresponding definitions of all types not coming from diesel or rust std-lib
* The corresponding table! definition from your schema.rs file
* The corresponding sql definition if the underlying tabl
* Information about the used rust and diesel version
* Information about your dbms and your operating system.

Without these information its unfortunately not possible to help you.

**Reply by jiangxiaoqiang** [reply]
> Please provide at least the following information:
> 
> * The corresponding definitions of all types not coming from diesel or rust std-lib
> * The corresponding table! definition from your schema.rs file
> * The corresponding sql definition if the underlying tabl
> * Information about the used rust and diesel version
> * Information about your dbms and your operating system.
> 
> Without these information its unfortunately not possible to help you.

this is the detail information: https://stackoverflow.com/questions/70547237/what-is-the-best-way-to-handle-the-id-when-insert-record-using-rust-diesel
